.github/workflows/build.yml

```
@@ -98,3 +98,63 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: test source - go${{ matrix.go-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        go-version:
+          - "1.19"
+          - "1.20"
+        include:
+          - os: ubuntu-20.04
+            go-version: "1.20"
+    steps:
+      - uses: actions/checkout@v3
+      - id: setup-go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-golang-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-golang-
```

I would like this to mirror how we do this in other skeletons.

```suggestion
        env:
          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
            go${{ steps.setup-go.outputs.go-version }}-"
        with:
          path: |
            ${{ steps.go-cache.outputs.dir }}
            ${{ steps.go-modcache.outputs.dir }}
          key: "${{ env.BASE_CACHE_KEY }}\
            ${{ hashFiles('**/go.sum') }}"
          restore-keys: |
            ${{ env.BASE_CACHE_KEY }}
```